src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -334,18 +334,25 @@ bool FollowMovementGenerator<T>::DoUpdate(T* owner, uint32 time_diff)
 
     target->GetNearPoint(owner, x, y, z, _range, 0.f, target->ToAbsoluteAngle(tAngle));
 
+    Movement::MoveSplineInit init(owner);
+
     bool success = i_path->CalculatePath(x, y, z, forceDest);
     if (!success || i_path->GetPathType() & PATHFIND_NOPATH)
     {
-        owner->StopMoving();
-        return true;
+        // Pet must always follow its master
+        if (followingMaster)
+            init.MoveTo(x, y, z);
+        else
+        {
+            owner->StopMoving();
+            return true;
+        }
```

Comment:
little codestyle improvement (instead of using else):

```
        if (!followingMaster) {
            owner->StopMoving();
            return true;
        }
        
        init.MoveTo(x, y, z);
  ```